config/routes.rb

```diff
@@ -4,4 +4,12 @@
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
   end
+
+  resources :posts, only: [:new, :create, :show] do
+    resources :comments, only: [:create]
+    member do
+      post 'like', to: 'likes#create'
+      delete 'unlike', to: 'likes#destroy'
+    end
+  end
```

Comment:
- [ ] These routes are not necessary since you already defined them on lines 4-6. Adding this 2nd resource outing brings about different post routes which is confusing. You can modify you're already existing resource routes to look like this:
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:new]
    end
  end
```